### PR TITLE
Security #2.5: Create builder for dial functions, add NewServer to tools

### PR DESCRIPTION
**Pull request by lobkovilya**
Signed-off-by: lobkovilya <ilya.lobkov@xored.com>

## Motivation and Context

- Add `tools.NewServer` function with opentracing support
- Add builder of dialFunc to simplify creating secure `Dial` and `NewServer`
## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controllers/sriov-controller/dpapi-controller.go [hunk 1]
```diff
@@ -177,7 +177,7 @@ func (s *serviceInstanceController) startServer() error {
 		return err
 	}
 
-	s.server = grpc.NewServer([]grpc.ServerOption{}...)
+	s.server = tools.NewServer()
```
Comment (nickolaev) [inline controllers/sriov-controller/dpapi-controller.go]: We really should deprecate this code somewhere away from the main tree. No point in maintaining something that we do not use.
@edwarnicke

File: pkg/tools/socket.go [hunk 2]
```diff
@@ -21,116 +22,151 @@ const (
...
+		return DialConfig{}, err
+	}
+
+	return rv, nil
 }
 
-// DialTCP simply calls DialTimeoutTCP with default timeout
-func DialTCP(address string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
-	return DialTimeoutTCP(address, dialTimeoutDefault, opts...)
+func readEnvBool(env string, value bool) (bool, error) {
```
Comment (nickolaev) [inline pkg/tools/socket.go]: This one should go to a more generic file.

File: pkg/tools/socket.go [hunk 3]
```diff
@@ -21,116 +22,151 @@ const (
...
+			otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer(), otgrpc.LogPayloads())),
+		grpc.WithStreamInterceptor(
+			otgrpc.OpenTracingStreamClientInterceptor(opentracing.GlobalTracer())),
+	}
 }
 
-// DialContextTCP establish TCP connection with address
-func DialContextTCP(ctx context.Context, address string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
-	addr, err := net.ResolveTCPAddr("tcp", address)
+func readConfiguration() (DialConfig, error) {
```
Comment (nickolaev) [inline pkg/tools/socket.go]: Maybe `readOpentracingConfiguration`?

File: pkg/tools/socket.go [hunk 4]
```diff
@@ -21,116 +22,151 @@ const (
...
+type dialContextFunc func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error)
 
-	if cfg.opentracing {
-		opts = append(opts,
-			grpc.WithUnaryInterceptor(
-				otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer(), otgrpc.LogPayloads())),
-			grpc.WithStreamInterceptor(
-				otgrpc.OpenTracingStreamClientInterceptor(opentracing.GlobalTracer())))
-	}
+type builder struct {
```
Comment (nickolaev) [inline pkg/tools/socket.go]: `dialBuilder`?

Comment (haiodo) [reply pkg/tools/socket.go]: +1

File: pkg/tools/socket.go (hunk 3 above)
Comment (haiodo) [reply pkg/tools/socket.go]: I suppose it will be not only opentracing but also security.

File: pkg/tools/socket.go (hunk 2 above)
Comment (haiodo) [reply pkg/tools/socket.go]: +1

**Pull request merged**